### PR TITLE
DEPLOY: use correct ARCH

### LINKED ISSUE #3782: macOS Monterey (aarch64) Illegal Instruction
If I download the macOS build in the Releases for v0.17.3.0 it crashes immediately upon launch with an illegal instruction exception. If I build the master branch with commit bddb9b0050ee086da5c7ee7fdb3627e0dff9f179 (HEAD -> master, tag: v0.17.3.0, origin/master, origin/HEAD) locally using Xcode 13.1 on macOS Monterey 12.0.1, everything works as expected. This is on a MacBook Pro 16 inch M1 Max. I've attached the crash log I get with the bundled version.

[CrashReport.txt](https://github.com/monero-project/monero-gui/files/7671748/CrashReport.txt)

### PR DESCRIPTION
**Pull request by selsta**
Closes #3782

### CODE REVIEW

File: DEPLOY.md [hunk 1]
```diff
@@ -8,11 +8,16 @@ Use macOS 10.12 - 10.13 for better backwards compability.
...
 
-4. `CMAKE_PREFIX_PATH=~/Qt5.12.8/5.12.8/clang_64 make release`
+4. Compile `monero-wallet-gui.app`
 
-5. `cd build/release && make deploy`
+```
+mkdir build && cd build
+cmake -D CMAKE_BUILD_TYPE=Release -D ARCH=default -D CMAKE_PREFIX_PATH=~/Qt5.12.8/5.12.8/clang_64 ..
+make
+make deloy
```
Comment (kpcyrd) [inline DEPLOY.md:17]:
```suggestion
make deploy
```

Comment (selsta) [reply DEPLOY.md]: Oops, thanks

**Comment by beshanoe** [comment]
should README.md be updated as well?

**Comment by selsta** [comment]
I don't see anything in README that needs updating.

**Comment by beshanoe** [comment]
ah, so for the local build for M1 I'm still using `make release -j4` and that's it?

**Comment by selsta** [comment]
correct

**Pull request merged**